packages/aws-cdk-lib/core/lib/removal-policy.ts

```diff
@@ -48,6 +48,19 @@ export enum RemovalPolicy {
    * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html#aws-attribute-deletionpolicy-options
    */
   SNAPSHOT = 'snapshot',
+
+  /**
+   * This uses the 'RetainExceptOnCreate' DeletionPolicy which behaves like Retain for stack operations,
+   * except for the stack operation that initially created the resource.
+   * When `applyToUpdateReplacePolicy` is set, this uses the 'Retain' UpdateReplacePolicy.
+   *
+   * If the stack operation that created the resource is rolled back, it will delete the resource.
+   * For all other stack operations, such as stack deletion, it retains the resource and its contents.
+   * The result is that new, empty, and unused resources are deleted, while in-use resources and their data are retained.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html#aws-attribute-deletionpolicy-options
+   */
+  RETAIN_EXCEPT_ON_CREATE= 'retain-except-on-create',
```

Comment:
All in all I think I'd like `RETAIN_EXISTING` or `RETAIN_AFTER_CREATE` better, but this will do